ReactAndroid/src/main/java/com/facebook/react/CoreModulesPackage.java

```diff
@@ -19,6 +19,7 @@
 import com.facebook.react.modules.core.DefaultHardwareBackBtnHandler;
 import com.facebook.react.modules.core.DeviceEventManagerModule;
 import com.facebook.react.modules.core.ExceptionsManagerModule;
+import com.facebook.react.modules.core.HMRClient;
```

Comment:
`com.facebook.react.modules.core.HMRClient` is from modules/core:core BUCK rule/package.

By importing this you create a circular dependency: modules/core:core -> devsupport:devsupport -> modules/core:core.
BUCK does not like circular dependencies.

So the easiest fix would be to move HMRClient class into devsupport package.